core-library/src/main/java/org/silverpeas/core/contribution/attachment/SimpleDocumentService.java

```diff
@@ -1042,6 +1042,23 @@ public void switchAllowingDownloadForReaders(final SimpleDocumentPK pk, final bo
     }
   }
 
+  @Override
+  public void switchEnableDisplayAsContent(final SimpleDocumentPK pk, final boolean enable) {
+    SimpleDocument document = searchDocumentById(pk, null);
+    final boolean documentUpdateRequired = enable != document.isDisplayableAsContent();
+
+    // Updating JCR if required
+    if (documentUpdateRequired) {
+      document.setDisplayableAsContent(enable);
+      try (JcrSession session = openSystemSession()) {
+        repository.saveDisplayableAsContent(session, document);
+        session.save();
+      } catch (RepositoryException ex) {
+        throw new AttachmentException(this.getClass().getName(), SilverpeasException.ERROR, "", ex);
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Remove this use of "SilverpeasException"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)